database/sqlite/sqlite_health.c

```diff
@@ -0,0 +1,1357 @@
+            ae->alarm_id == alarm_id && (!(ae->hash_name == hash_name && ae->hash_chart == hash_chart &&
+                                           !strcmp(name, ae->name) && !strcmp(chart, ae->chart))))) {
+            return 0;
+        }
+    }
+    return 1;
+}
+
+/* Health related SQL queries
+   Load from the health log table
+*/
+#define SQL_LOAD_HEALTH_LOG(guid,limit) "SELECT hostname, unique_id, alarm_id, alarm_event_id, config_hash_id, updated_by_id, updates_id, when_key, duration, non_clear_duration, flags, exec_run_timestamp, delay_up_to_timestamp, name, chart, family, exec, recipient, source, units, info, exec_code, new_status, old_status, delay, new_value, old_value, last_repeat, class, component, type FROM (SELECT hostname, unique_id, alarm_id, alarm_event_id, config_hash_id, updated_by_id, updates_id, when_key, duration, non_clear_duration, flags, exec_run_timestamp, delay_up_to_timestamp, name, chart, family, exec, recipient, source, units, info, exec_code, new_status, old_status, delay, new_value, old_value, last_repeat, class, component, type FROM health_log_%s order by unique_id desc limit %u) order by unique_id asc;", guid, limit
+void sql_health_alarm_log_load(RRDHOST *host) {
+    sqlite3_stmt *res = NULL;
+    int rc;
+    char command[MAX_HEALTH_SQL_SIZE + 1];
+
+    host->health_log_entries_written = 0;
+
+    if (unlikely(!db_meta)) {
+        if (default_rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE)
+            error_report("HEALTH [%s]: Database has not been initialized", host->hostname);
+        return;
+    }
+
+    char uuid_str[GUID_LEN + 1];
+    uuid_unparse_lower(host->host_uuid, uuid_str);
+    uuid_str[8] = '_';
+    uuid_str[13] = '_';
+    uuid_str[18] = '_';
+    uuid_str[23] = '_';
+
+    snprintfz(command, MAX_HEALTH_SQL_SIZE, SQL_LOAD_HEALTH_LOG(uuid_str, host->health_log.max));
+
+    rc = sqlite3_prepare_v2(db_meta, command, -1, &res, 0);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("HEALTH [%s]: Failed to prepare sql statement to load health log.", host->hostname);
+        return;
+    }
+
+    netdata_rwlock_rdlock(&host->health_log.alarm_log_rwlock);
+
+    while (sqlite3_step(res) == SQLITE_ROW) {
+        errno = 0;
+        ssize_t errored = 0;
+
+        ALARM_ENTRY *ae = NULL;
+
+        // check that we have valid ids
+        uint32_t unique_id = (uint32_t) sqlite3_column_int64(res, 1);
+        if(!unique_id) {
+            error_report("HEALTH [%s]: Got invalid unique id. Ignoring it.", host->hostname);
+            errored++;
+            continue;
+        }
+
+        uint32_t alarm_id = (uint32_t) sqlite3_column_int64(res, 2);
+        if(!alarm_id) {
+            error_report("HEALTH [%s]: Got invalid alarm id. Ignoring it.", host->hostname);
+            errored++;
+            continue;
+        }
+
+        // Check if we got last_repeat field
+        time_t last_repeat = 0;
+        char* alarm_name = strdupz((char *) sqlite3_column_text(res, 14));
```

Comment:
need to allocate memory? (and free it in L755?)

Reply:
Will use the the sqlite_column_text by itself, thanks!